w3-filecoin.md

```diff
@@ -0,0 +1,187 @@
+
+### Authority
+
+_Authority_ is a [principal] that executes invoked capabilities.
+
+### Verifier
+
+Component of the [authority] that performs UCAN validation
+
+# Protocol
+
+## Overview
+
+A Storage Provider is the entry point for user/application data into the web3. It will act on behalf of users to move data around into different storage points. One of the key storage points is Filecoin deals.
+
+## Authorization
+
+Storage Provider (web3.storage) must be given capabilities by a Filecoin broker Authority (Spade) to invoke capabilities to get its stored content into Filecoin deals.
+
+```mermaid
+sequenceDiagram
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+    participant Storage Provider as 🌐<br/><br/>did:web:web3.storage #32;
+
+    Authority->>Storage Provider: delegate `filecoin/add`
+```
+
+## Filecoin deals
+
+When a storage provider has enough 
+
+```mermaid
+sequenceDiagram
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+    participant Storage Provider as 🌐<br/><br/>did:web:web3.storage #32;
+
+    Storage Provider->>Authority: invoke `filecoin/add`
+    Note left of Authority: Prepare for deal
+    Authority-->>Storage Provider: receipt
+    Note left of Authority: Land deal in Filecoin
+    Authority->>Storage Provider: invoke `filecoin/deal`
+    Authority-->>Storage Provider: invoke `filecoin/fail`
+```
+
+## Capabilities
+
+This section describes the capabilities that form the w3 Filecoin protocol, along with details relevant to invoking capabilities with a service provider.
+
+### `filecoin/add`
+
+A Filecoin broker Authority MAY delegate capabilities to any storage provider.
+
+> `did:web:spade.storage` delegates `filecoin/add` capabilities to `did:web:web3.storage`
+
+```json
+{
+  "iss": "did:web:spade.storage",
+  "aud": "did:web:web3.storage",
+  "att": [{ "with": "did:web:spade.storage", "can": "filecoin/add" }],
+  "exp": null,
+  "sig": "..."
+}
+```
+
+> `did:web:web3.storage` invokes capability from `did:web:spade.storage`
+
+```json
+{
+  "iss": "did:web:web3.storage",
+  "aud": "did:web:spade.storage",
+  "att": [{
+    "with": "did:web:spade.storage",
+    "can": "filecoin/add",
+    "nb": : {
+      "aggregateId": "id...",
+      "cars": [
+        {
+          "link": "bag...",
+          "size": 1234,
+          "md5": "md5...",
+          "commP": "commP...",
+          "url": "https://..."
+        }
+      ]
+    }
+  }],
+  "prof": [
+    "iss": "did:web:spade.storage",
+    "aud": "did:web:web3.storage",
+    "att": [{ "with": "did:web:spade.storage", "can": "filecoin/add" }],
+    "exp": null,
+    "sig": "..."
+  ],
+  "sig": "..."
+}
+```
+
+### `filecoin/deal`
+
+A Filecoin broker authority MAY invoke deal capability when one aggregate lands into a Filecoin deal. It should include the CID of the request CAR cid from `filecoin/add` invocation.
```

Comment:
As mentioned earlier I think it's best to leverage effects and have general interface for subscribing for the updates. But assuming we choose to not use them here, I would suggest using a different namespace on the web3.storage side because:

1. I would like to be able to say `web3.storage` implements `filecoin/*` protocol which can not be true if some capabilities are part of the other end of the pipeline. In other words I think it's good idea to have `/add` be in the different namespace from `/deal` and `/fail`